genesyscloud/team/resource_genesyscloud_team.go

```diff
@@ -0,0 +1,147 @@
+	}
+
+	d.SetId(*teamObj.Id)
+	log.Printf("Created team %s", *teamObj.Id)
+	return readTeam(ctx, d, meta)
+}
+
+// readTeam is used by the team resource to read an team from genesys cloud
+func readTeam(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTeamProxy(sdkConfig)
+
+	log.Printf("Reading team %s", d.Id())
+
+	return gcloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		team, respCode, getErr := proxy.getTeamById(ctx, d.Id())
+		if getErr != nil {
+			if gcloud.IsStatus404ByInt(respCode) {
+				return retry.RetryableError(fmt.Errorf("Failed to read team %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read team %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceTeam())
+
+		resourcedata.SetNillableValue(d, "name", team.Name)
+
+		resourcedata.SetNillableReferenceWritableDivision(d, "division_id", team.Division)
+
+		resourcedata.SetNillableValue(d, "description", team.Description)
+		resourcedata.SetNillableValue(d, "member_count", team.MemberCount)
+
+		log.Printf("Read team %s %s", d.Id(), *team.Name)
+		return cc.CheckState()
+	})
+}
+
+// updateTeam is used by the team resource to update an team in Genesys Cloud
+func updateTeam(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTeamProxy(sdkConfig)
+
+	team := getTeamFromResourceData(d)
+
+	log.Printf("Updating team %s", *team.Name)
+	teamObj, err := proxy.updateTeam(ctx, d.Id(), &team)
+	if err != nil {
+		return diag.Errorf("Failed to update team: %s", err)
+	}
+
+	log.Printf("Updated team %s", *teamObj.Id)
+	return readTeam(ctx, d, meta)
+}
+
+// deleteTeam is used by the team resource to delete an team from Genesys cloud
+func deleteTeam(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTeamProxy(sdkConfig)
+
+	_, err := proxy.deleteTeam(ctx, d.Id())
+	if err != nil {
+		return diag.Errorf("Failed to delete team %s: %s", d.Id(), err)
+	}
+
+	return gcloud.WithRetries(ctx, 180*time.Second, func() *retry.RetryError {
+		_, respCode, err := proxy.getTeamById(ctx, d.Id())
+
+		if err != nil {
+			if gcloud.IsStatus404ByInt(respCode) {
+				log.Printf("Deleted team %s", d.Id())
+				return nil
+			}
+			return retry.NonRetryableError(fmt.Errorf("Error deleting team %s: %s", d.Id(), err))
+		}
+
+		return retry.RetryableError(fmt.Errorf("team %s still exists", d.Id()))
+	})
+}
+
+// getTeamFromResourceData maps data from schema ResourceData object to a platformclientv2.Team
+func getTeamFromResourceData(d *schema.ResourceData) platformclientv2.Team {
+
+	name := d.Get("name").(string)
+	division := d.Get("division_id").(string)
+	//description := d.Get("description").(string)
```

Comment:
Why is description commented out?  Is this a field we want to include?

Reply:
its included in couple of lines below.